app/src/main/java/dev/shreyaspatil/foodium/data/local/dao/PostsDao.kt

```diff
@@ -59,6 +60,8 @@ interface PostsDao {
     @Query("SELECT * FROM ${Post.TABLE_NAME} WHERE ID = :postId")
     fun getPostById(postId: Int): Flow<Post>
 
+    fun getPostByIdUntilChanged(id: Int) = getPostById(id).distinctUntilChanged()
```

Comment:
I think no need to change this in DAO instead I'll suggest call `distinctUntilChanged()` from repository layer itself. Can you please do that change?

Reply:
Ok. will do that